### PR TITLE
Refactor RNG to use collections for state management

### PR DESCRIPTION
**Pull request by sethrj**

### CODE REVIEW

File: src/random/RngInterface.hh [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+{
+    // TODO: std::shared_ptr<std::mt19937> rng; ?
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * Initialize an RNG.
+ */
+template<MemSpace M>
+struct RngIntializer;
```
Comment (amandalund) [inline src/random/RngInterface.hh:98]: This guy is misspelled throughout.

**Review by amandalund — COMMENTED** [review]
Just one tiny fix -- looks great Seth!

File: src/random/RngInterface.hh (hunk 1 above)
Comment (sethrj) [reply src/random/RngInterface.hh:98]: Good catch! That would have been embarrassing 😂

**Review by amandalund — APPROVED** [review]
Thanks @sethrj!

**Pull request merged**